modules/automatadBidAdapter.js

```diff
@@ -0,0 +1,123 @@
+import {registerBidder} from '../src/adapters/bidderFactory.js'
+import * as utils from '../src/utils.js'
+import {BANNER} from '../src/mediaTypes.js'
+import {ajax} from '../src/ajax.js'
+
+const BIDDER = 'automatad'
+
+const ENDPOINT_URL = 'https://rtb2.automatad.com/ortb2'
+
+const DEFAULT_BID_TTL = 30
+const DEFAULT_CURRENCY = 'USD'
+const DEFAULT_NET_REVENUE = true
+
+export const spec = {
+  code: BIDDER,
+  aliases: ['atd'],
+  supportedMediaTypes: [BANNER],
+
+  isBidRequestValid: function (bid) {
+    // will receive request bid. check if have necessary params for bidding
+    return (bid && bid.hasOwnProperty('params') && bid.params.hasOwnProperty('siteId') && bid.params.hasOwnProperty('placementId') && bid.hasOwnProperty('mediaTypes') && bid.mediaTypes.hasOwnProperty('banner'))
+  },
+
+  buildRequests: function (validBidRequests, bidderRequest) {
+    if (!validBidRequests || !bidderRequest) {
+      return
+    }
+
+    const siteId = validBidRequests[0].params.siteId
+    const placementId = validBidRequests[0].params.placementId
+
+    const impressions = validBidRequests.map(bidRequest => ({
+      id: bidRequest.bidId,
+      banner: {
+        format: bidRequest.sizes.map(sizeArr => ({
+          w: sizeArr[0],
+          h: sizeArr[1],
+        }))
+      },
+    }))
+
+    // params from bid request
+    const openrtbRequest = {
+      id: validBidRequests[0].auctionId,
+      imp: impressions,
+      site: {
+        id: siteId,
+        placement: placementId,
+        domain: window.location.hostname,
+        page: window.location.href,
+        ref: bidderRequest.refererInfo ? bidderRequest.refererInfo.referer || null : null,
+      },
+    }
+
+    const payloadString = JSON.stringify(openrtbRequest)
+    return {
+      method: 'POST',
+      url: ENDPOINT_URL + '/resp',
+      data: payloadString,
+      options: {
+        contentType: 'application/json',
+        withCredentials: false,
+        crossOrigin: true,
+      },
+    }
+  },
+
+  interpretResponse: function (serverResponse, request) {
+    const bidResponses = []
+    const response = (serverResponse || {}).body
+
+    if (response && response.seatbid && response.seatbid.length === 1 && response.seatbid[0].bid && response.seatbid[0].bid.length) {
+      response.seatbid[0].bid.forEach(bid => {
+        bidResponses.push({
+          requestId: bid.impid,
+          cpm: bid.price,
+          ad: bid.adm,
+          adDomain: bid.adomain[0],
```

Comment:
Can you fix to meta.advertiserDomains?

Reply:
@patmmccann I have created a pull request for this change #6509